controllers/usersignup/usersignup_controller.go

```diff
@@ -106,6 +106,20 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	}
 	logger = logger.WithValues("username", userSignup.Spec.Username)
 
+	// TODO remove this section (and the referenced function) after migration has completed
+	// FROM HERE ---------
+	migrated, err := r.migrateUserSignupClaimsIfNecessary(ctx, userSignup)
```

Comment:
Let's move it after checking if the UserSignup resource is being deleted. There is no point of migrating the UserSignup which is being deleted.

Reply:
Done.